Sources/Client/Client/Client.swift

```diff
@@ -126,6 +129,21 @@ public final class Client: Uploader {
         self.logOptions = logOptions
         logger = logOptions.logger(icon: "🐴", for: [.requestsError, .requests, .requestsInfo])
         
+        // Seetup a default WebSocketProvider type.
+        if let defaultWebSocketProviderType = defaultWebSocketProviderType {
+            self.defaultWebSocketProviderType = defaultWebSocketProviderType
+        } else if #available(iOS 13, *) {
+            // Use iOS build-in WebSocket provider.
+            // Starscream does the same by default.
+            self.defaultWebSocketProviderType = URLSessionWebSocketProvider.self
+        } else if let providerClass = NSClassFromString("StreamChatClient.StarscreamWebSocketProvider"),
+            let starscreamWebSocketProviderClass = providerClass as? WebSocketProvider.Type {
+            // Use Starscream WebSocket provider lazily.
+            self.defaultWebSocketProviderType = starscreamWebSocketProviderClass
+        } else {
+            self.defaultWebSocketProviderType = nil
+        }
```

Comment:
Isn't it possible to remove these runtime checks and make it all compile-time safe? Something like:
```swift
#if canImport(Starscream)

#elseif canImport(URLSessionWebSocketTask)

#else
    #error("Starscream is required for systems where URLSessionWebSocketTask is not available.")
#endif
```
So can't even compile when Starscrem is missing and the target is <iOS13. (not sure if this is actually possible)